src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobIncrementalCopy.cs

```diff
@@ -0,0 +1,455 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+namespace Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet
+{
+    using Commands.Common.Storage.ResourceModel;
+    using Microsoft.WindowsAzure.Commands.Common.Storage;
+    using Microsoft.WindowsAzure.Commands.Storage.Common;
+    using Microsoft.WindowsAzure.Commands.Storage.Model.Contract;
+    using Microsoft.WindowsAzure.Storage;
+    using Microsoft.WindowsAzure.Storage.Blob;
+    using System;
+    using System.Management.Automation;
+    using System.Security.Permissions;
+    using System.Threading.Tasks;
+
+    [Cmdlet(VerbsLifecycle.Start, StorageNouns.IncrementalCopyBlob, SupportsShouldProcess = true, DefaultParameterSetName = ContainerParameterSet),
+       OutputType(typeof(AzureStorageBlob))]
+    public class StartAzureStorageBlobIncrementalCopy : StorageCloudBlobCmdletBase
+    {
+        private const string BlobTypeMismatch = "Blob type of the blob reference doesn't match blob type of the blob.";
+
+        /// <summary>
+        /// Blob Pipeline parameter set name
+        /// </summary>
+        private const string BlobParameterSet = "BlobInstance";
+
+        /// <summary>
+        /// Blob Pipeline parameter set name
+        /// </summary>
+        private const string BlobToBlobParameterSet = "BlobInstanceToBlobInstance";
+
+        /// <summary>
+        /// Container pipeline paremeter set name
+        /// </summary>
+        private const string ContainerParameterSet = "ContainerInstance";
+
+        /// <summary>
+        /// Blob name and container name parameter set
+        /// </summary>
+        private const string ContainerNameParameterSet = "ContainerName";
+
+        /// <summary>
+        /// Source uri parameter set
+        /// </summary>
+        private const string UriParameterSet = "UriPipeline";
+
+        [Alias("SrcICloudBlob", "SrcCloudBlob", "ICloudBlob")]
+        [Parameter(HelpMessage = "CloudBlob Object", Mandatory = true, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = BlobParameterSet)]
+        [Parameter(HelpMessage = "CloudBlob Object", Mandatory = true, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = BlobToBlobParameterSet)]
+        public CloudPageBlob CloudBlob { get; set; }
+
+        [Parameter(HelpMessage = "CloudBlobContainer Object", Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = ContainerParameterSet)]
+        public CloudBlobContainer CloudBlobContainer { get; set; }
+
+        [Parameter(HelpMessage = "Blob name", ParameterSetName = ContainerParameterSet, Mandatory = true)]
+        [Parameter(HelpMessage = "Blob name", ParameterSetName = ContainerNameParameterSet, Mandatory = true)]
+        public string SrcBlob
```

Comment:
@blueww this applies to all cmdlets that have the `Src` and `Dest` prefix:

Can you change the parameter name to include the full word for `Src` and `Dest`? For example, `SrcFoo` will be renamed to `SourceFoo`, and you can add an alias to `SrcFoo` so users can still use it. Same goes for `DestFoo` and `DestinationFoo`.